src/Thesis/QueryRunner.hs

```diff
@@ -0,0 +1,25 @@
+module Thesis.QueryRunner where
+
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.Catch (MonadMask)
+import Data.Time.Units (TimeUnit)
+import Database.PostgreSQL.Simple (Connection)
+import System.Random (StdGen)
+
+import Thesis.Composition.PrivacyFilter
+import Thesis.Mechanism (laplace)
+import Thesis.Query
+import Thesis.ValueGuard
+
+run :: (TimeUnit t, MonadIO m, MonadMask m, PrivacyFilter p) => StdGen -> Connection -> p -> Query -> t -> m (p, StdGen, Either BudgetDepleted Double)
+run gen conn privacyFilter query timeout =
+  case nonNegative 0 of
```

Comment:
Can we maybe add a helper function in the `ValueGuard` module to construct `delta = 0`?